java/src/main/native/src/RmmJni.cpp

```diff
@@ -577,6 +578,80 @@ class pinned_fallback_host_memory_resource {
                            cuda::mr::host_accessible) noexcept
   {
   }
+
+#if CCCL_MAJOR_VERSION > 3 || (CCCL_MAJOR_VERSION == 3 && CCCL_MINOR_VERSION >= 1)
```

Comment:
Please also move this `#if...#endif` up, after line 548 so they can stay closer to the similar functions.